contracts/LeverageEngine.sol

```diff
@@ -64,14 +66,27 @@ contract LeverageEngine is AccessController {
     ///
     /// @param ousdPrinciple the amount of OUSD sent to Archimedes
     /// @param cycles How many leverage cycles to do
-    function createLeveragedPosition(uint256 ousdPrinciple, uint256 cycles) external expectInitialized nonReentrant returns (uint256) {
+    /// @param archAmount Arch tokens to burn for position
+    function createLeveragedPosition(
+        uint256 ousdPrinciple,
+        uint256 cycles,
+        uint256 archAmount
+    ) external expectInitialized nonReentrant returns (uint256) {
         uint256 lvUSDAmount = _parameterStore.getAllowedLeverageForPosition(ousdPrinciple, cycles);
-        _leverageAllocator.useAvailableLvUSD(msg.sender, lvUSDAmount);
+        uint256 lvUSDAmountAllocatedFromArch = _parameterStore.calculateLeverageAllowedForArch(archAmount);
+        /// Revert if not enough Arch token for needed leverage. Continue if too much arch is given
+        require(lvUSDAmountAllocatedFromArch >= lvUSDAmount, "Not enough Arch provided");
+
+        uint256 availableLev = _coordinator.getAvailableLeverage();
+        require(availableLev >= lvUSDAmount, "Not enough available lvUSD");
+
+        _burnArchTokenForPosition(msg.sender, archAmount);
         uint256 positionTokenId = _positionToken.safeMint(msg.sender);
         _ousd.safeTransferFrom(msg.sender, _addressCoordinator, ousdPrinciple);
         _coordinator.depositCollateralUnderNFT(positionTokenId, ousdPrinciple);
         _coordinator.getLeveragedOUSD(positionTokenId, lvUSDAmount);
 
+        emit PositionCreated(msg.sender, positionTokenId, ousdPrinciple, lvUSDAmount, archAmount);
```

Comment:
do we want to also emit here the arch price ratio for the round?

Reply:
Don't know if its needed in the event(trying to keep is small for lower gas fees). It can also be derived from lvUSDAmount + archAmount

Reply:
yeah good point